lib/plaindate.ts

```diff
@@ -351,6 +350,7 @@ export class PlainDate implements Temporal.PlainDate {
       calendar
     );
     const instant = ES.BuiltinTimeZoneGetInstantFor(timeZone, dt, 'compatible');
+    if (!instant) throw new TypeError(`Could not create instant for timezone "${timeZone}" and dateTime "${dt}"`);
```

Comment:
Can `instant` actually be undefined here?  I don't think it can be. The type problem is  upstream. For example, `DisambiguatePossibleInstants` needs an assertion at the end to prevent `undefined` from being returned, e.g. 

```js
    throw new Error(`assertion failed: invalid disambiguation value ${disambiguation}`);
```